bam_sort.c

```diff
@@ -1934,7 +1934,7 @@ static int bam_merge_simple(SamOrder sam_order, char *sort_tag, const char *out,
     }
 
     if (sam_close(fpout) < 0) {
-        print_error(cmd, "error closing output file");
+        print_error_errno(cmd, "error closing output file \"%s\"", fn[i]);
```

Comment:
out would be the appropriate filename instead of fn[I] as fpout is opened using it.